tests/cli/test_compose_tar_kickstart.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
+# Note: execute this file from the project root directory
+
+#####
+#
+# Build tar image and install it using liveimg kickstart command
+#
+#####
+
+set -e
+
+. /usr/share/beakerlib/beakerlib.sh
+. $(dirname $0)/lib/lib.sh
+
+CLI="${CLI:-./src/bin/composer-cli}"
+
+rlJournalStart
+    rlPhaseStartSetup
+        rlAssertExists $QEMU_BIN
+        if ! rlCheckRpm httpd; then
+            yum -y install httpd
+        fi
+        systemctl start httpd
+
+        ks_path="/var/www/html/ks-tar.cfg"
+        tmp_dir=$(mktemp -d /tmp/composer.XXXXX)
+        ssh_key_dir=$(mktemp -d /tmp/composer-ssh-keys.XXXXXX)
+
+        rlRun -t -c "ssh-keygen -t rsa -N '' -f $ssh_key_dir/id_rsa"
+        pub_key=$(cat $ssh_key_dir/id_rsa.pub)
+
+        bp_name="test-tar"
+        blueprint="$bp_name.toml"
+        cat > $blueprint << __EOF__
+name = "$bp_name"
+description = "tar image test"
+version = "0.0.1"
+modules = []
+
+[[packages]]
+name = "openssh-server"
+version = "*"
+
+# sudo and auditd are needed for checks performed on the installed image instance
+[[packages]]
+name = "sudo"
+version = "*"
+
+[[packages]]
+name = "audit"
+version = "*"
+
+[[groups]]
+name = "anaconda-tools"
+version = "*"
+
+[[customizations.user]]
+name = "root"
+key = "$pub_key"
+
+__EOF__
+        rlRun -t -c "$CLI blueprints push $blueprint"
+        image_path="/var/www/html/root.tar.xz"
+
+        version=$(awk -F = '$1 == "VERSION_ID" { print $2 }' /etc/os-release | tr -d \")
+        arch=$(uname -m)
+        baseurl="http://download.eng.bos.redhat.com/rel-eng/latest-RHEL-${version}/compose/BaseOS/x86_64/os/"
+        rlRun -t -c "curl --remote-name-all $baseurl/images/pxeboot/{vmlinuz,initrd.img}"
+
+        rlRun -t -c "fallocate -l 5G disk.img"
+    rlPhaseEnd
+
+    rlPhaseStartTest "compose start"
+        rlAssertEquals "SELinux operates in enforcing mode" "$(getenforce)" "Enforcing"
+        uuid=$($CLI compose start $bp_name tar)
+        rlAssertEquals "exit code should be zero" $? 0
+
+        uuid=$(echo $uuid | cut -f 2 -d' ')
+    rlPhaseEnd
+
+    rlPhaseStartTest "compose finished"
+        if [ -n "$uuid" ]; then
+            until $CLI compose status | grep -E "$uuid (FINISHED|FAILED)"; do
```

Comment:
Missing `\` before `|`. (Same error also on rhel7-extras branch).

Reply:
I fail to see why the backslash should be put there? If it was just `grep` without `-E`, I'd need to escape the parentheses and pipe, but as it is an extended regexp, this is not the case. Or is it something else you are getting at?

Reply:
Hmm, I didn't actually try to see if the backslash was required. I just noticed that was different from the other code snippets that reference the same code. There is actually a helper function but if you want to use it then you'll have to cherry-pick it from master which is outside the scope of this change. 

You can still do it if you want. IMO having the test suite not diverge too much between branches is a good thing.

Reply:
You're right that we should strive for the lowest divergence possible across branches. I had a look at the state of the aforementioned piece of code checking for compose finish on master is different from `rhel7-extras` and `rhel8-branch`, plus that the function you mentioned hasn't been backported from master. I totally agree that using the `wait_for_compose` function is the best approach, however, at this state, I'd rather consider backporting this in different PRs (both for `rhel7-extras` and `rhel8-branch`) later on.

Reply:
While testing https://github.com/weldr/lorax/pull/887 on Fedora 31 the compose itself failed and b/c it wasn't using the `wait_for_compose()` function we didn't have logs. I've made the necessary change and will open tickets to cherry-pick these changes for the rhel branches.